Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1111,5 +1111,105 @@ public void When_asserting_a_document_has_a_null_element_it_should_fail()
         }
 
         #endregion
+
+        #region HaveSingleElement
+
+        [Fact]
+        public void When_asserting_document_has_a_single_child_element_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                  </parent>");
+
+            // Act / Assert
+            document.Should().HaveSingleElement("child");
+        }
+
+        [Fact]
+        public void When_asserting_document_has_single_child_element_but_it_does_have_two_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act
+            Action act = () => document.Should().HaveSingleElement("child");
```

Comment:
You could also validate that `because` and `becauseArgs` are properly passed along in this test.
See e.g. `XDocumentAssertionSpecs.When_the_expected_element_is_null_it_fails`